modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/certificatevalidation/CertificateVerificationManager.java

```diff
@@ -0,0 +1,282 @@
+
+                        log.debug("Valid issuer certificate found in the client truststore. Caching..");
+
+                        // Store the valid issuer cert in cache for future use
+                        certCache.setCacheValue(peerCert.getSerialNumber().toString(), issuerCert);
+                        if (log.isDebugEnabled()) {
+                            log.debug("Issuer certificate with serial number: " + issuerCert.getSerialNumber()
+                                    .toString() + " has been cached against the serial number:  " + peerCert
+                                    .getSerialNumber().toString() + " of the peer certificate.");
+                        }
+                        break;
+                    } catch (SignatureException | CertificateException | NoSuchAlgorithmException |
+                             InvalidKeyException | NoSuchProviderException e) {
+                        // Unable to verify the signature. Check with the next certificate in the next loop traversal.
+                    }
+                }
+            } else {
+                X509Certificate cachedIssuerCert = certCache.getCacheValue(peerCert.getSerialNumber().toString());
+                try {
+                    peerCert.verify(cachedIssuerCert.getPublicKey());
+                } catch (SignatureException | CertificateException | NoSuchAlgorithmException |
+                         InvalidKeyException |
+                         NoSuchProviderException e) {
+                    // Unable to verify the signature.
+                    throw new CertificateVerificationException("Unable to verify the signature of the certificate.");
+                }
+            }
+        }
+
+        long start = System.currentTimeMillis();
+
+        OCSPCache ocspCache = OCSPCache.getCache();
+        ocspCache.init(cacheSize, cacheDelayMins);
+        CRLCache crlCache = CRLCache.getCache();
+        crlCache.init(cacheSize, cacheDelayMins);
+
+        RevocationVerifier[] verifiers = {new OCSPVerifier(ocspCache), new CRLVerifier(crlCache)};
+
+        for (RevocationVerifier verifier : verifiers) {
+            try {
+                if (isFullCertChainValidationEnabled) {
+
+                    if (isCertExpiryValidationEnabled) {
+                        log.debug("Validating certificate chain for expiry");
+                        if (isExpired(convertedCertificates)) {
+                            throw new CertificateVerificationException("One of the provided certificates are expired");
+                        }
+                    }
+
+                    log.debug("Doing full certificate chain validation");
+                    CertificatePathValidator pathValidator = new CertificatePathValidator(convertedCertificates,
+                            verifier);
+                    pathValidator.validatePath();
+                    log.info("Path verification Successful. Took " + (System.currentTimeMillis() - start) + " ms.");
```

Comment:
This info log is not required.

Reply:
Removed with https://github.com/wso2/wso2-synapse/pull/2119/commits/09b48b314fa878b08f5287ee7ab2026ac5f29a01